core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/random.c

```diff
@@ -49,19 +49,20 @@ random_buf(void *buf, size_t len)
 
 #elif defined(BH_PLATFORM_WINDOWS)
 
-#include <wincrypt.h>
+#include <bcrypt.h>
+#pragma comment(lib, "Bcrypt.lib")
+#include <ntstatus.h>
 
 void
 random_buf(void *buf, size_t len)
 {
-    static int crypt_initialized = 0;
-    static HCRYPTPROV provider;
-    if (!crypt_initialized) {
-        CryptAcquireContext(&provider, NULL, NULL, PROV_RSA_FULL,
-                            CRYPT_VERIFYCONTEXT);
-        crypt_initialized = 1;
+    NTSTATUS ret = BCryptGenRandom(NULL, buf, (ULONG)len,
+                         BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+    if (BCryptGenRandom(NULL, buf, (ULONG)len,
+                         BCRYPT_USE_SYSTEM_PREFERRED_RNG)) {
+        os_printf("BCryptGenRandom failed");
+        abort();
```

Comment:
What was the purpose for this change?

Reply:
Opened a separate PR for that, https://github.com/bytecodealliance/wasm-micro-runtime/pull/2769 - I only noticed when compiling WAMR with uwp toolchain since `CryptGenRandom` isn't available on UWP and then when I opened the windows docs, I noticed it was deprecated and there's a newer version which is recommended so thought I might as well fix it.